freedom-ldscript-generator.c++

```diff
@@ -154,11 +154,21 @@ static void dts_memory (void)
     int dtim_count = 0;
     int testram_count = 0;
     int spi_count = 0;
+    int memory_count = 0;
     int periph_count = 0;
     int sys_count = 0;
 
     auto dtb = fdt((const uint8_t *)dts_blob);
     dtb.match(
+        std::regex("memory"), [&](node n) {
+            auto name = n.name();
+            n.named_tuples(
```

Comment:
Named tuples is designed to handle the "reg", "reg-names" pairs, I think there's already a function to handle the unnamed tuples that'll fit better.